src/core/models/outputs/output_extended.c

```diff
@@ -240,8 +238,26 @@ output_extended_t* output_extended_deserialize(byte_t buf[], size_t buf_len) {
     offset += sizeof(uint16_t);
   }
 
+  // unlock condition blocks
+  uint8_t unlock_count = 0;
+  memcpy(&unlock_count, &buf[offset], sizeof(uint8_t));
+  if (unlock_count == 0 || unlock_count > MAX_EXTENDED_CONDITION_BLOCKS_COUNT) {
+    printf("[%s:%d] invalid unlock block count\n", __func__, __LINE__);
+    output_extended_free(output);
+    return NULL;
+  } else {
+    output->unlock_conditions = cond_blk_list_deserialize(buf + offset, buf_len - offset);
+    if (!output->unlock_conditions) {
+      printf("[%s:%d] can not deserialize unlock conditions\n", __func__, __LINE__);
+      output_extended_free(output);
+      return NULL;
+    }
+    offset += cond_blk_list_serialize_len(output->unlock_conditions);
+  }
+
   // feature blocks
-  uint8_t feat_block_count = *((uint8_t*)&buf[offset]);
+  uint8_t feat_block_count = 0;
+  memcpy(&feat_block_count, &buf[offset], sizeof(uint8_t));
   if (feat_block_count > 0) {
```

Comment:
Is here missing additional check for `MAX_EXTENDED_FEATURE_BLOCKS_COUNT`? Like it is in for unlock condition block (line 244)?
```suggestion
  if (feat_block_count > 0 || feat_block_count <= MAX_EXTENDED_FEATURE_BLOCKS_COUNT) {
```